ops/terraform/aws/network/main.tf

```diff
@@ -0,0 +1,204 @@
+
+  tags {
+    Name = "${var.application_name}: app subnet ${count.index}"
+    env  = "${var.environment}"
+    app  = "${var.application_name}"
+  }
+}
+
+# DMZ subnets
+resource "aws_subnet" "dmz" {
+  count             = "${length(var.availability_zones)}"
+  vpc_id            = "${aws_vpc.application_vpc.id}"
+  cidr_block        = "${element(var.az_cidr_blocks["dmz"], count.index)}"
+  availability_zone = "${var.region}${element(var.availability_zones, count.index)}"
+
+  tags {
+    Name = "${var.application_name}: dmz subnet ${count.index}"
+    env  = "${var.environment}"
+    app  = "${var.application_name}"
+  }
+}
+
+# Security Group: world -> alb
+resource "aws_security_group" "application_alb_sg" {
+  name   = "${var.application_name}: ${var.environment} public alb security group"
+  vpc_id = "${aws_vpc.application_vpc.id}"
+
+  ingress {
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"
+    cidr_blocks = ["${var.public_ingress_cidrs}"]
+  }
+
+  ingress {
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = ["${var.public_ingress_cidrs}"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags {
+    Name = "${var.application_name}: ${var.environment} public alb security group"
+    env  = "${var.environment}"
+    app  = "${var.application_name}"
+  }
+}
+
+# Security group: alb -> app subnet
+resource "aws_security_group" "application_app_sg" {
+  name   = "${var.application_name}: ${var.environment} application subnet security group"
+  vpc_id = "${aws_vpc.application_vpc.id}"
+
+  ingress {
+    from_port       = "${var.application_port}"
+    to_port         = "${var.application_port}"
+    protocol        = "tcp"
+    security_groups = ["${aws_security_group.application_alb_sg.id}"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags {
+    Name = "${var.application_name}: ${var.environment} application subnet security group"
+    env  = "${var.environment}"
+    app  = "${var.application_name}"
+  }
+}
+
+# Public alb
+resource "aws_alb" "application_alb" {
+  name            = "${var.application_name}-${var.environment}"
+  internal        = false
+  security_groups = ["${aws_security_group.application_alb_sg.id}"]
+  subnets         = ["${aws_subnet.dmz.*.id}"]
+
+  ip_address_type = "ipv4"
+
+  tags {
+    Name = "${var.application_name}: ${var.environment} alb"
+    env  = "${var.environment}"
+    app  = "${var.application_name}"
+  }
+}
+
+resource "aws_alb_target_group" "application_target_group" {
+  name     = "${var.application_name}-${var.environment}"
+  port     = "${var.application_port}"
+  protocol = "HTTP"
+  vpc_id   = "${aws_vpc.application_vpc.id}"
+
+  health_check {
+    interval            = 60
+    path                = "/_health"
```

Comment:
We should make this a TF variable I think -- we might want to standardize on a healthcheck URL path that all apps conform to, but this should be overridable.